bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/OpenAPIUtils.java

```diff
@@ -66,28 +78,42 @@ public static Request requestBuilder(HttpClient httpClient, NanoleafControllerCo
         }
         URI requestURI;
         try {
-            requestURI = new URI(HttpScheme.HTTP.asString(), null, address, port, path, null, null);
+            requestURI = new URI(HttpScheme.HTTP.asString(), null, address, port, path, query, null);
         } catch (URISyntaxException use) {
-            logger.warn("URI could not be parsed with path {}", path);
+            LOGGER.warn("URI could not be parsed with path {}", path);
             throw new NanoleafException("Wrong URI format for API request");
         }
-        return httpClient.newRequest(requestURI).method(method);
+        return requestURI;
     }
 
     public static ContentResponse sendOpenAPIRequest(Request request)
             throws NanoleafException, NanoleafUnauthorizedException {
         try {
+            LOGGER.trace("Sending Request {} {}",request.getURI(), request.getQuery() == null ? "no query parameters": request.getQuery());
+            LOGGER.trace("Request method:{} uri:{} params{}\n", request.getMethod(), request.getURI(), request.getParams());
+            if (request.getContent()!=null) {
+                Iterator<ByteBuffer> iter = request.getContent().iterator();
```

Comment:
We normally don't user iterator's directly are you sure this is the way here?

Reply:
I think there is no other way to retrieve the request content except using the iterator. See also https://stackoverflow.com/questions/33264411/jetty-9-httpclient-how-to-get-the-request-content

Reply:
If it is in a specific debug method then I'm okay with it. However note you just put that content in it by yourself, so there might be more optimal places to log it :-)

Reply:
yes, it is now in its own "traceSendRequest"-method. I know that we use it ourselves but the intent was to have central place where can trace all requests.